addons/project/models/project.py

```diff
@@ -0,0 +1,653 @@
+    date_last_stage_update = fields.Datetime('Last Stage Update', index=True, copy=False, readonly=True, default=fields.Datetime.now)
+    project_id = fields.Many2one('project.project', 'Project', ondelete='set null', index=True, track_visibility='onchange',
+        change_default=True, default=lambda self: self.env.context.get('default_project_id'))
+    notes = fields.Text()
+    planned_hours = fields.Float('Initially Planned Hours', help='Estimated time to do the task, usually set by the project manager when the task is in draft state.')
+    remaining_hours = fields.Float('Remaining Hours', digits=(16,2), help="Total remaining time, can be re-estimated periodically by the assignee of the task.")
+    user_id = fields.Many2one('res.users', 'Assigned to', index=True, track_visibility='onchange', default=lambda self: self.env.user)
+    partner_id = fields.Many2one('res.partner', 'Customer', default=_get_default_partner)
+    manager_id = fields.Many2one('res.users', related='project_id.user_id', string='Project Manager')
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.user.company_id)
+    color = fields.Integer('Color Index')
+    user_email = fields.Char(related='user_id.email', string='User Email', readonly=True)
+    attachment_ids = fields.One2many('ir.attachment', 'res_id', domain=lambda self: [('res_model', '=', self._name)], auto_join=True, string='Attachments')
+    # In the domain of displayed_image_id, we couln't use attachment_ids because a one2many is represented as a list of commands so we used res_model & res_id
+    displayed_image_id = fields.Many2one('ir.attachment', domain="[('res_model', '=', 'project.task'), ('res_id', '=', id), ('mimetype', 'ilike', 'image')]", string='Displayed Image')
+    legend_blocked = fields.Char(related="stage_id.legend_blocked", string='Kanban Blocked Explanation')
+    legend_done = fields.Char(related="stage_id.legend_done", string='Kanban Valid Explanation')
+    legend_normal = fields.Char(related="stage_id.legend_normal", string='Kanban Ongoing Explanation')
+
+
+    @api.constrains('date_start','date_end')
+    def _check_dates(self):
+        for task in self:
+            if task.date_start and task.date_end and task.date_start > task.date_end:
+                raise UserError(_("Error ! Task starting date must be lower than its ending date."))
+
+    @api.onchange('remaining_hours')
+    def onchange_remaining(self):
+        if not self.planned_hours:
+            self.planned_hours = self.remaining_hours
+
+    @api.onchange('planned_hours')
+    def onchange_planned(self):
+        self.remaining_hours = self.planned_hours
+
+    @api.onchange('project_id')
+    def onchange_project_id_wrapper(self):
+        result = self.onchange_project(self.project_id.id)['value']
+        for fname, value in result.iteritems():
+            setattr(self, fname, value)
+
+    @api.multi
+    def onchange_project(self, project_id):
```

Comment:
@mdi-odoo I make this wrapper method because onchange_project() is override from 'project_timesheet' module.